# Warn about overcommit_memory after daemonising so it is not spammed to the
# console on boot.
#
#  -- Chris Lamb <lamby@debian.org>  Wed, 17 Jun 2009 10:27:44 +0100

diff -urNad /tmp/bp-build/redis-0.900.orig/redis.c /tmp/bp-build/redis-0.900/redis.c
--- redis-0.900.orig/redis.c	2009-06-17 10:24:06.000000000 +0100
+++ redis-0.900/redis.c	2009-06-17 10:24:38.000000000 +0100
@@ -4432,10 +4432,6 @@
 }
 
 int main(int argc, char **argv) {
-#ifdef __linux__
-    linuxOvercommitMemoryWarning();
-#endif
-    
     initServerConfig();
     if (argc == 2) {
         ResetServerSaveParams();
@@ -4449,6 +4445,9 @@
     initServer();
     if (server.daemonize) daemonize();
     redisLog(REDIS_NOTICE,"Server started, Redis version " REDIS_VERSION);
+#ifdef __linux__
+    linuxOvercommitMemoryWarning();
+#endif
     if (rdbLoad(server.dbfilename) == REDIS_OK)
         redisLog(REDIS_NOTICE,"DB loaded from disk");
     if (aeCreateFileEvent(server.el, server.fd, AE_READABLE,
